cpp/lib/Authenticator.cpp

```diff
@@ -194,6 +212,10 @@ extern "C" {
       {
```

Comment:
I think better to use
if (conn->auth_object)
{
    delete static_cast<Snowflake::Client::IAuthenticator*>(conn->auth_object);
}

Reply:
Updated.